### PR TITLE
Add mypy roadmap

### PR DESCRIPTION
**Pull request by JukkaL**

### CODE REVIEW

**Comment by gvanrossum** [comment]
Thanks! Are you also deleting the defunct one from the mypy-lang website? Or adding a link back to this file?

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+- Make error messages more useful and informative.
+  ([issue](https://github.com/python/mypy/labels/topic-usability))
+
+- Resolve [#2008](https://github.com/python/mypy/issues/2008) (we are
+  converging on approach 4).
+
+## July-December 2017
+
+- Invest some effort into systematically filling in missing
```
Comment (JelleZijlstra) [inline ROADMAP.md]: As you're probably aware, I'm already trying to work on this. My current approach is pretty haphazard, so let me know if you want to coordinate on filling in specific areas.

**Comment by JukkaL** [comment]
I'll add a link to the new roadmap to the mypy website. The old roadmap is already gone (see http://mypy-lang.org/roadmap.html).

File: ROADMAP.md (hunk 1 above)
Comment (JukkaL) [inline ROADMAP.md]: The idea we have is to implement an analysis that can help us find the stubs (or parts of stubs) that cause the largest number of `Any` types when type checking a program. We haven't started work on this yet, but we'll keep you updated once we make some progress on this.

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  ([issue](https://github.com/python/mypy/issues/1240))
+
+- Add support for statically typed
+  [protobufs](https://developers.google.com/protocol-buffers/).
+
+- Provide much faster, reliable interactive feedback through
+  fine-grained incremental type checking, built on top the daemon
+  mode.
+
+- Start work on editor plugins and support for selected IDE features.
```
Comment (ilevkivskyi) [inline ROADMAP.md]: Does this roadmap support proposals for addition of items?
I have few such proposals for July-December 2017 section:
* Make strict-optional the default
* Better support for union types (e.g. in overloads, see https://github.com/python/mypy/issues/3295, in joins etc.)
* Better support for forward references (there are currently several issues about this). This is 90% dependent on re-working semantic analysis passes

And maybe also (for later time?):
* Better support for algebraic data types (aka ``NamedUnion``, see https://github.com/python/mypy/issues/2464)
* Simple dependent types (https://github.com/python/mypy/issues/3062 and https://github.com/python/mypy/issues/3345)

Comment (gvanrossum) [inline ROADMAP.md]: I think we do want to move forward with strict-optional. For the other issues, I'm not sure these warrant roadmap inclusion -- we don't want the roadmap to become synonymous with the issue tracker, and we're not saying we want to ignore the issue tracker.

**Pull request merged**